screenpy/pacing.py

```diff
@@ -87,7 +125,7 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
             completed_line = f"{line.format(actor, **cues)}"
             with the_narrator.stating_a_beat(func, completed_line, gravitas) as n_func:
                 retval = n_func(*args, **kwargs)
-                if retval is not None:
+                if retval is not None or function_attached_to_protocol(n_func):
```

Comment:
Does this work? `n_func` is the function that `beat` annotated wrapped in the Narrator's context(s); we could just do `func` here instead directly which is the function that `beat` is annotating.

Reply:
It does work, but you're totally right it should be just `func` for clarity sake.